synapse/storage/push_rule.py

```diff
@@ -185,6 +185,46 @@ def bulk_get_push_rules(self, user_ids):
 
         defer.returnValue(results)
 
+    @defer.inlineCallbacks
+    def copy_push_rules_from_room_to_room_for_user(self, old_room_id, new_room_id, user_id):
+        """Copy the push rules from one room to another for a specific user.
+
+        Args:
+            old_room_id (str): ID of the old room.
+            new_room_id (str): ID of the new room.
+            user_id (str): ID of user to copy push rules for.
+        """
+        # Retrieve push rules
+        user_push_rules = yield self.get_push_rules_for_user(user_id)
+
+        # Get rules for the old room
+        for rule in user_push_rules:
+            if old_room_id in rule["rule_id"]:
```

Comment:
is this really the right way to check for which room a rule applies to? surely the id is meant to be opaque?

Reply:
The ID is `global/room/<room id>`, but we can use the pattern instead.